src/Sarif.Multitool/Rules/SARIF2009.UseConventionalSymbolicNames.cs

```diff
@@ -0,0 +1,105 @@
+        private static readonly Regex s_conventionalIdRegex = new Regex(@"^[A-Z]{1,5}[0-9]{1,4}$", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        protected override void Analyze(Run run, string runPointer)
+        {
+            if (run.OriginalUriBaseIds != null)
+            {
+                AnalyzeOriginalUriBaseIds(run.OriginalUriBaseIds, runPointer.AtProperty(SarifPropertyName.OriginalUriBaseIds));
+            }
+        }
+
+        protected override void Analyze(Tool tool, string toolPointer)
+        {
+            if (tool.Driver != null)
+            {
+                AnalyzeToolDriver(tool.Driver, toolPointer.AtProperty(SarifPropertyName.Driver));
+            }
+        }
+
+        private void AnalyzeToolDriver(ToolComponent toolComponent, string toolDriverPointer)
+        {
+            if (toolComponent.Rules != null)
+            {
+                string rulesPointer = toolDriverPointer.AtProperty(SarifPropertyName.Rules);
+                for (int i = 0; i < toolComponent.Rules.Count; i++)
+                {
+                    AnalyzeReportingDescriptor(toolComponent.Rules[i], rulesPointer.AtIndex(i));
+                }
+            }
+        }
+
+        private void AnalyzeReportingDescriptor(ReportingDescriptor reportingDescriptor, string reportingDescriptorPointer)
+        {
+            if (string.IsNullOrWhiteSpace(reportingDescriptor.Id))
+            {
+                return;
+            }
+
+            if (!s_conventionalIdRegex.IsMatch(reportingDescriptor.Id))
+            {
+                // {0}: The 'id' property of the rule '{1}' does not follow the recommended format:
+                // a short string identifying the tool concatenated with a numeric rule number, for example, `CS2001`.
+                // Using a conventional format for the rule id provides a more uniform experience across tools.
+                LogResult(
+                    reportingDescriptorPointer.AtProperty(SarifPropertyName.Id),
+                    nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text),
+                    reportingDescriptor.Name);
+            }
+        }
+
+        private void AnalyzeOriginalUriBaseIds(IDictionary<string, ArtifactLocation> originalUriBaseIds, string originalUriBaseIdsPointer)
+        {
+            foreach (KeyValuePair<string, ArtifactLocation> originalUriBaseId in originalUriBaseIds)
+            {
+                if (!s_conventionalSymbols.Contains(originalUriBaseId.Key))
+                {
+                    // {0}: The 'originalUriBaseIds' symbol '{1}' is not one of the conventional symbols. 
+                    // We suggest 'REPOROOT' for the root of a repository, 'SRCROOT' for the root of the directory containing all source code,
+                    // 'TESTROOT' for the root of the directory containing all test code (if your repository is organized in that way),
```

Comment:
the lines of this comment are too long and need to be truncated.

Reply:
just updated! Can you check again? Thank you

---
In reply to: [445553388](https://github.com/microsoft/sarif-sdk/pull/1938#discussion_r445553388) [](ancestors = 445553388)

Reply:
Improved! My rule of thumb, though, is that there should never be complete words that are to the right margin of the shortest preceding line. On line 100, for example, 'directory' should have been dropped to the next line (as it is entirely right of the margin established on line 99 by 'symbols.' It is fine for a word to 'poke' past the end of the margin, as long as it starts before that rightmost margin. Make sense? Sorry to nit-pick so much on the comments, but rules source code is authored once and truly read many many times as people use a tool, are confused by results, want to understand more, etc. Analysis rules like this also are effectively a set of guidelines encoded in C#. At some point, these rules might be reauthored in another analysis environment. 

---
In reply to: [445569283](https://github.com/microsoft/sarif-sdk/pull/1938#discussion_r445569283) [](ancestors = 445569283,445553388)

Reply:
i will keep your guidelines in mind! Thanks for sharing and reviewing. just updated!

---
In reply to: [445585630](https://github.com/microsoft/sarif-sdk/pull/1938#discussion_r445585630) [](ancestors = 445585630,445569283,445553388)

Reply:
+1 on Michael's formatting guidelines.

---
In reply to: [445591404](https://github.com/microsoft/sarif-sdk/pull/1938#discussion_r445591404) [](ancestors = 445591404,445585630,445569283,445553388)